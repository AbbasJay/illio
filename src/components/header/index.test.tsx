// test component renders
// test the dateTo prop is given to the first input value
// test the dateFrom prop is given to the first input value
// test when the dateTo value is changed the setDateTo function is called
// test when the dateFrom value is changed the setDateFrom function is called

import React from "react";
import { mount } from "enzyme";

import renderer from "react-test-renderer";

import Header from "./";

const props = {
  ticker: "MCD",
  dateFrom: "2020-01-01",
  dateTo: "2020-01-05",
  setDateFrom: () => jest.fn(),
  setDateTo: () => jest.fn(),
};

const renderedComponent = () => {
  return mount(<Header {...props} />);
};

describe("Header", () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it("renders correctly to snapshot", () => {
    const snapshot = renderer.create(<Header {...props} />).toJSON();

    expect(snapshot).toMatchSnapshot();
  });

  it("should render component", () => {
    const component = renderedComponent();

    expect(component.length).toBe(1);
  });

  describe("props", () => {
    it("should have the correct ticker prop", () => {
      const component = renderedComponent();

      expect(component.prop("ticker")).toEqual(props.ticker);
    });

    it("should have the correct dateTo prop", () => {
      const component = renderedComponent();

      expect(component.prop("dateTo")).toEqual(props.dateTo);
    });

    it("should have the correct dateFrom prop", () => {
      const component = renderedComponent();

      expect(component.prop("dateFrom")).toEqual(props.dateFrom);
    });

    // it("should setDateFrom prop when onChange is triggered", () => {
    //   const component = renderedComponent();

    //   component.find(".date-from-input").simulate("change");

    //   expect(props.setDateFrom).toHaveBeenCalled();
    // });

    // it("should setDateTo prop when onChange is triggered", () => {
    //   const component = renderedComponent();

    //   component.find(".date-to-input").simulate("change");

    //   expect(props.setDateTo).toHaveBeenCalled();
    // });
  });
});
