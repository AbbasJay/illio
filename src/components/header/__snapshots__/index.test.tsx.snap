// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly to snapshot 1`] = `
<div
  className="header"
>
  <div
    className="header__ticker"
  >
    <span>
      Ticker
    </span>
    <input
      readOnly={true}
      value="MCD"
    />
  </div>
  <div
    className="header__from"
  >
    <span>
      From
    </span>
    <input
      className="date-from-input"
      onChange={[Function]}
      type="date"
      value="2020-01-01"
    />
  </div>
  <div
    className="header__to"
  >
    <span>
      To
    </span>
    <input
      className="date-to-input"
      onChange={[Function]}
      type="date"
      value="2020-01-05"
    />
  </div>
</div>
`;
